src/geometry/geometry.cpp

```diff
@@ -58,6 +51,22 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   geometry->AddField(geometric_variables::cell_coords, gcoord_cell);
   geometry->AddField(geometric_variables::node_coords, gcoord_node);
 
+  // Reductions
+  Params &params = geometry->AllParams();
+  if (params.hasKey("xh")) {
+    auto HstSum = parthenon::UserHistoryOperation::sum;
+    using History::ReduceMassAccretionRate;
+    using parthenon::HistoryOutputVar;
+    parthenon::HstVar_list hst_vars = {};
+
+    auto ReduceAccretionRate = [](MeshData<Real> *md) {
+      return ReduceMassAccretionRate<Kokkos::Sum<Real>>(md);
+    };
+
+    hst_vars.emplace_back(HistoryOutputVar(HstSum, ReduceAccretionRate, "mdot"));
+    params.Add(parthenon::hist_param_key, hst_vars);
+  }
```

Comment:
Hmm it occurs to me that something gross about putting this in `geometry::Initialize` is that this machinery is present even if the fluid variables are not. (One could imagine running, e.g., radiation around a black hole, for example.)

One can resolve this (in a kind of hacky way) by also checking the following boolean condition from the input deck
```C++
bool hydro_on = pin->GetBoolean("physics", "hydro");
```